### DISCUSSION TITLE
Global metadata should not be needed in data repositories

### DISCUSSION THREAD

**Discussion by brockfanning** [open-sdg/open-sdg#1442] [Ideas]
Right now the contents of the "Global metadata" tab are fields that must be in the data repository's "meta" files. But this is all official UN metadata and does not need to be custom or controlled. Is there some way that we can leverage UN (and/or other) sources to populate the "Global metadata" tab?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I think having this would make having a translated site quite a bit easier as it would be one less thing that countries would have to translate.

So is the first step of this getting the global metadata translated?

Would all of the translations then go in to sdg-translations and that repo would be pointed at instead of the metadata files for global metadata?

**Comment by brockfanning** [comment]
When I proposed this, yes that's what I was thinking: all the global metadata could go into sdg-translations, and then we'd update open-sdg so that the global metadata was pulled from there, rather than relying on indicator metadata fields.

I still think that's a good general direction, but I believe that the UN is making a push to translate the global metadata. I think we'll know more specifics about this in the next few months. So it might be a good idea to hold off on this for now, and revisit when we have a more solid idea of what the authoritative source of global metadata will look like.

That said, in a way though, we've already started on this, and could continue. The open-sdg platform is still relying on indicator metadata fields for the global metadata, but in some cases we are "pointing" to sdg-translations by virtue of using a translation key. For example, see the 'indicator_name' field in the starter repo, [here](https://raw.githubusercontent.com/open-sdg/open-sdg-data-starter/develop/meta/1-1-1.md). It's pointing to "global_indicators.1-1-1-title", which is a roundabout way of using the global translation.

Perhaps an easy next step would be to do the same with the "un_custodian_agency" field, for example by translating "World Bank" and adding it to sdg-translations, and then changing the data starter repo to something like this:
```
un_custodian_agency: custodian_agencies.world_bank
```

If we wanted to do that, we would need to do this:
1. Compile a list of all the custodian agencies in the global indicators
2. Translate them into all the languages we have on sdg-translations (this would be the hardest part)
3. Add those to sdg-translations as a new file called 'custodian_agencies.yml'
4. Update open-sdg-data-starter to point to translation keys (such as "custodian_agencies.world_bank")

**Comment by LucyGwilliamAdmin** [comment]
Although looking at sdg-translations most of the translations are already there.
Global metadata fields:

| Field | In sdg-translations |
|------|----------------------|
|Indicator name|Yes|
|Indicator number|Not needed?|
|Target name|Yes|
|Target number|Not needed?|
|Global indicator description|No|
|UN designated tier|Not needed?|
|UN custodian agency|No|
|Link to UN metadata|No|

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning sorry I had some more thoughts so was just looking into a bit more.

So in the table above, I think we need 3 fields translated which aren't already - I'm not sure if numbers stay the same in different alphabets but those should be ok to translate without any help if required.

I don't think there is a huge list of custodian agencies - the translated versions should be somewhere I imagine.

And as for what goes in the Link to UN metadata field, this again is something that would need to be translated just once for each language

I think the biggest thing stopping this going forward at the moment is the Global indicator description field.

Yes, I agree we could wait but depending on how easy/difficult it is to point at the translations instead of the metadata files maybe we could start moving it over?

**Comment by LucyGwilliamAdmin** [comment]
Also is Global indicator description part of the global metadata?

I'm just looking at the [global metadata file for 1.2.1](https://unstats.un.org/sdgs/metadata/files/Metadata-01-02-01.pdf) and can't see which part this would be and this is something we don't seem to have filled in in the UK.

**Comment by brockfanning** [comment]
Regarding the description field, I'm not sure either. It seems like there is the potential for a lot of fields ("Comments and Limitations", etc.) If I had to guess I'd say the description could be what is "Definition" under "Concepts and definitions".

After thinking more I think there's not a big reason to wait on the shorter fields. The custodian agency seems like a nice "low hanging fruit", since it should be pretty easy to translate. And the "Link to UN metadata" doesn't even require a translator, so would be another easy one.

For the longer fields like description (whatever that is) the translation is going to involve more human translator involvement, and so I'm a little more inclined to wait and see what the UN produces over the next few months. That said, the UN is only ever going to support the 6 official languages, so it would certainly be fine to move forward with the translations of indicator descriptions (or any other metadata) in any other languages.

The only gotcha there is that it remains to be seen how the UN will "chunk up" the metadata - in other words, how they will divide all that metadata into fields. I would assume that each section in the PDF would be it's own field (eg, Definition, Rationale, Concepts, etc.) but I could be wrong.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning 
I'm thinking since some indicators have more than one custodian agency do we want the set up to be:
```
translations:
  lang:
    global_custodian_agencies.yml:
      1-1-1-custodian-agencies: xxx
      1-2-1-custodian-agencies: xxx
```
and then in the metadata file `un_custodian_agencies: [INDICATOR-CODE]-custodian-agencies`?

**Comment by brockfanning** [comment]
That works for me, if you think that's preferable.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @AnnCorp I'm having a conversation with someone in our data team about making sure that the tier and and link to global metadata link is always up-to-date. So even though, these don't need to be translated they would need to be present in each language folder within translations-un-sdg. Getting these into un-sdg-translations shouldn't be too much work, but I'm just thinking about how we would keep them up-to-date when the tier is changed?

Update: actually don't think we would want them to be in individual translation folders as not something that need to be translated but maybe somewhere else within open-sdg, where they can be pulled from for every site.

**Comment by LucyGwilliamAdmin** [comment]
There's this [file which lists all the changes](https://unstats.un.org/sdgs/files/List_of_changes_since_15_Oct_2018.pdf) made to tiers, custodian agencies, etc. so I guess one way to keep up-to-date would be to check that link every so often.

**Comment by LucyGwilliamAdmin** [comment]
This can be closed once starter repo is pointing at UN API
